### PR TITLE
Send SIGKILL after SIGTERM when passing 95% memory

### LINKED ISSUE #6373: Flaky `test_slow_terminate`
[`test_slow_terminate`](https://github.com/dask/distributed/blob/4d292461ecf4a33a78547b94648064051fd757d3/distributed/tests/test_worker_memory.py#L762-L790) failed unexpectedly in https://github.com/dask/distributed/pull/6372 with a standard 30s test timeout

```
                    ) from None
E                   asyncio.exceptions.TimeoutError: Test timeout after 30s.
E                   ========== Test stack trace starts here ==========
E                   Stack for <Task pending name='Task-172714' coro=<test_slow_terminate() running at /home/runner/work/distributed/distributed/distributed/tests/test_worker_memory.py:788> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7f802626fd90>()]>> (most recent call last):
E                     File "/home/runner/work/distributed/distributed/distributed/tests/test_worker_memory.py", line 788, in test_slow_terminate
E                       await leak_until_restart(c, s, a)
```

https://github.com/dask/distributed/runs/6503645519?check_suite_focus=true

There are tons of warnings like

```
2022-05-19 09:01:54,693 - distributed.worker_memory - WARNING - Unmanaged memory use is high. This may indicate a memory leak or the memory may not be released to the OS; see https://distributed.dask.org/en/latest/worker-memory.html#memory-not-released-back-to-the-os for more information. -- Unmanaged memory: 6.14 GiB -- Worker memory limit: 400.00 MiB
```

cc @crusaderky

### PR DESCRIPTION
**Pull request by crusaderky**
Closes #6373

CC @hendrikmakait

### CODE REVIEW

File: distributed/worker_memory.py [hunk 1]
```diff
@@ -344,21 +345,40 @@ def memory_monitor(self, nanny: Nanny) -> None:
...
-        if process.pid in (self._last_terminated_pid, None):
-            # We already sent SIGTERM to the worker, but its handler is still running
-            # since the previous iteration of the memory_monitor - for example, it
-            # may be taking a long time deleting all the spilled data from disk.
+        if memory / self.memory_limit <= self.memory_terminate_fraction:
             return
-        self._last_terminated_pid = -1
 
-        if memory / self.memory_limit > self.memory_terminate_fraction:
+        if self._last_terminated_pid != process.pid:
```
Comment (fjetter) [inline distributed/worker_memory.py:351]: The default interval is 100ms. Isn't this a bit short for us to escalate to kill?

Comment (crusaderky) [reply distributed/worker_memory.py:351]: Not really - that's 100ms worth of extra leaking from any task that is currently running.

Note that, to the best of my understanding, no cleanup code whatsoever runs on SIGTERM by default. So it should be immediate, unless the user tampered with the signal handlers - which they could legitimately do for their own cleanup code. A realistic use case is a database library that installs a SIGTERM handler to cleanly shut down its sockets.

In this case, I argue that prompt termination for a process that is close to go beyond 100% is more important the clean teardown.

**Comment by crusaderky** [comment]
Ready for review and merge

File: distributed/worker_memory.py (hunk 1 above)
Comment (fjetter) [reply distributed/worker_memory.py:354]: We do install our own signal handlers and close workers/nannies, e.g.

https://github.com/dask/distributed/blob/7665eaacf0780594c33670e75b3517380102263f/distributed/_signals.py#L11

https://github.com/dask/distributed/blob/7665eaacf0780594c33670e75b3517380102263f/distributed/cli/dask_worker.py#L479-L487

Comment (crusaderky) [reply distributed/worker_memory.py:354]: I know. This doesn't exclude third party handlers.
Typical pattern:
```python

def my_handler(signo, frame):
    # TODO do your thing
    prev(signo, frame)

prev = signal.signal(signal.SIGTERM, my_handler)
```

**Review by hendrikmakait — APPROVED** [review]
LGTM!

**Pull request merged**